### ISSUE TITLE
[Filebeat] Bridge legacy Event Hub SDK logs to logp

### ISSUE THREAD

**Issue by zmoog** [elastic/beats#40449]
## Context

The legacy Azure Event Hub SDK uses the package https://github.com/devigned/tab tracing and logs.

From [https://github.com/Azure/azure-event-hubs-go/../storage/storage.go#L285-L289](https://github.com/Azure/azure-event-hubs-go/blob/3dc3aeb6144e5d79a70d32f023f9d5990f81569d/storage/storage.go#L285-L289):

```go
	lease, err := sl.getLease(ctx, partitionID)
	if err != nil {
		tab.For(ctx).Error(err) // <-------- 👀
		return nil, false, nil
	}
```

## Problem

However, this package only supports OpenTracing or OpenCensus, which Beats doesn't use. Without OpenTracing or OpenCensus, the `github.com/devigned/tab` defaults to the `NoOpTracer`, so we do not have logs for the Event Hub SDK.

From [https://github.com/devigned/tab/../trace.go#L7-L9](https://github.com/devigned/tab/blob/355fbdc61144b946aed5ba021d0ba1e9e286844b/trace.go#L7-L9)

```go
var (
	tracer Tracer = new(NoOpTracer)
)
```

## Solutions

To get visibility into the Event Hub SDK, we need to bridge logs between https://github.com/devigned/tab and the github.com/elastic/elastic-agent-libs/logp package Beats uses for logging.

**Comment by zmoog** [comment]
The `github.com/devigned/tab` package seems in use only by the following:

```shell
$ go mod why -m github.com/devigned/tab

# github.com/devigned/tab
github.com/elastic/beats/v7/x-pack/filebeat/input/azureeventhub
github.com/Azure/azure-event-hubs-go/v3
github.com/devigned/tab
```

So building a custom tracer to bridge the logs from `github.com/Azure/azure-event-hubs-go/v3` to `github.com/elastic/elastic-agent-libs/logp` should not harm any other Beats component.

**Comment by zmoog** [comment]
Quick test using https://github.com/elastic/beats/pull/40451, and I can get a lot of details about the processor and the leaser/checkpointer:

```shell
$ pbpaste | grep '^{' | jq -r 'select(."log.origin"."file.name" == "azureeventhub/tracer.go") | [."@timestamp",."log.level",.message] | @tsv'

2024-08-07T16:17:12.305+0200	debug	successful rpc on link 494e91ef-8e06-4460-b89b-361d44c64f75: status code 200 and description:
2024-08-07T16:17:17.306+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb": running scan
2024-08-07T16:17:19.667+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb": acquired: [{"partitionID":"1","epoch":21,"owner":"86edc94f-dd34-44d3-8a43-ecf3cf4c77fb","checkpoint":{"offset":"201863710488","sequenceNumber":8568,"enqueueTime":"2024-08-07T13:35:51.837Z"},"state":"available","token":"32ab10ca-cc08-42a8-83fd-1dc436f25abc"} {"partitionID":"0","epoch":23,"owner":"86edc94f-dd34-44d3-8a43-ecf3cf4c77fb","checkpoint":{"offset":"206158444904","sequenceNumber":8565,"enqueueTime":"2024-08-07T13:26:15.225Z"},"state":"available","token":"e4aa4ecd-ab0f-4984-92cd-a4c6dbcfa433"}], not acquired: []
2024-08-07T16:17:19.667+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "1", epoch 21: running...
2024-08-07T16:17:19.667+0200	debug	creating a new receiver
2024-08-07T16:17:21.349+0200	debug	negotiating claim for audience amqps://mbranca-general.servicebus.windows.net/activitylogs/ConsumerGroups/$Default/Partitions/1 with token type servicebus.windows.net:sastoken and expiry of 1723047441
2024-08-07T16:17:21.562+0200	debug	successful rpc on link 3f9e7e50-f012-4692-8172-8b8d5c9dde17: status code 202 and description: Accepted
2024-08-07T16:17:21.563+0200	debug	negotiated with response code 202 and message: Accepted
2024-08-07T16:17:22.293+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "0", epoch 23: running...
2024-08-07T16:17:22.293+0200	debug	creating a new receiver
2024-08-07T16:17:24.705+0200	debug	negotiating claim for audience amqps://mbranca-general.servicebus.windows.net/activitylogs/ConsumerGroups/$Default/Partitions/0 with token type servicebus.windows.net:sastoken and expiry of 1723047445
2024-08-07T16:17:25.013+0200	debug	successful rpc on link 6bb4086b-b8ae-46ca-87dc-ef4e18c15db2: status code 202 and description: Accepted
2024-08-07T16:17:25.013+0200	debug	negotiated with response code 202 and message: Accepted
2024-08-07T16:17:30.261+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "1", epoch 21: lease renewed
2024-08-07T16:17:32.620+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "0", epoch 23: lease renewed
2024-08-07T16:17:35.807+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb": running scan
2024-08-07T16:17:36.566+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb": acquired: [], not acquired: [{"partitionID":"0","epoch":23,"owner":"86edc94f-dd34-44d3-8a43-ecf3cf4c77fb","checkpoint":{"offset":"206158444904","sequenceNumber":8565,"enqueueTime":"2024-08-07T13:26:15.225Z"},"state":"leased","token":"e4aa4ecd-ab0f-4984-92cd-a4c6dbcfa433"} {"partitionID":"1","epoch":21,"owner":"86edc94f-dd34-44d3-8a43-ecf3cf4c77fb","checkpoint":{"offset":"201863710488","sequenceNumber":8568,"enqueueTime":"2024-08-07T13:35:51.837Z"},"state":"leased","token":"32ab10ca-cc08-42a8-83fd-1dc436f25abc"}]
2024-08-07T16:17:43.233+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "0", epoch 23: lease renewed
2024-08-07T16:17:46.407+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb": running scan
2024-08-07T16:17:47.382+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb": acquired: [], not acquired: [{"partitionID":"0","epoch":23,"owner":"86edc94f-dd34-44d3-8a43-ecf3cf4c77fb","checkpoint":{"offset":"206158444904","sequenceNumber":8565,"enqueueTime":"2024-08-07T13:26:15.225Z"},"state":"leased","token":"e4aa4ecd-ab0f-4984-92cd-a4c6dbcfa433"} {"partitionID":"1","epoch":21,"owner":"86edc94f-dd34-44d3-8a43-ecf3cf4c77fb","checkpoint":{"offset":"201863710488","sequenceNumber":8568,"enqueueTime":"2024-08-07T13:35:51.837Z"},"state":"leased","token":"32ab10ca-cc08-42a8-83fd-1dc436f25abc"}]
2024-08-07T16:17:50.100+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "1", epoch 21: lease renewed
2024-08-07T16:17:52.965+0200	debug	eph "86edc94f-dd34-44d3-8a43-ecf3cf4c77fb", partition "0", epoch 23: lease renewed
```
